ursabot/tests/test_reporters.py

```diff
@@ -0,0 +1,107 @@
+# flake8: noqa  todo removeme
+from buildbot.process.results import FAILURE, SUCCESS, EXCEPTION
+from buildbot.test.unit import test_reporter_github as original
+
+from ursabot.reporters import (GitHubStatusPush, GitHubReviewPush,
+                               GitHubCommentPush)
+from ursabot.utils import ensure_deferred
+
+
+class TestGitHubStatusPush(original.TestGitHubStatusPush):
+
+    def setService(self):
+        # test or own implementation
+        self.sp = GitHubStatusPush(token='XXYYZZ')
+        return self.sp
+
+
+class TestGitHubStatusPushURL(original.TestGitHubStatusPushURL):
+    # project must be in the form <owner>/<project>
+    TEST_PROJECT = 'buildbot'
+    TEST_REPO = 'https://github.com/buildbot1/buildbot1.git'
+
+    def setService(self):
+        # test or own implementation
+        self.sp = GitHubStatusPush(token='XXYYZZ')
+        return self.sp
+
+
+class TestGitHubReviewPush(TestGitHubStatusPush):
+
+    def setService(self):
+        self.sp = GitHubReviewPush(token='XXYYZZ')
+        return self.sp
+
+    @ensure_deferred
+    async def test_basic(self):
+        build = await self.setupBuildResults(SUCCESS)
+
+        self._http.expect(
+            'post',
+            '/repos/buildbot/buildbot/pulls/34/reviews',
+            json={
+                'event': '',
+                'commit_id': 'd34db33fd43db33f'
```

Comment:
Where does this commit id come from?

Reply:
From [here](https://github.com/buildbot/buildbot/blob/ec38d8f7bbffa4fc792d5bb6466c336015131197/master/buildbot/test/util/reporter.py#L23)